genesyscloud/recording_media_retention_policy/resource_genesyscloud_recording_media_retention_policy.go

```diff
@@ -0,0 +1,207 @@
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	pp := getPolicyProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	order := d.Get("order").(int)
+	description := d.Get("description").(string)
+	enabled := d.Get("enabled").(bool)
+	mediaPolicies := buildMediaPolicies(d, pp, ctx)
+	conditions := buildConditions(d)
+	actions := buildPolicyActionsFromResource(d, pp, ctx)
+	policyErrors := buildPolicyErrors(d)
+
+	reqBody := platformclientv2.Policycreate{
+		Name:          &name,
+		Order:         &order,
+		Description:   &description,
+		Enabled:       &enabled,
+		MediaPolicies: mediaPolicies,
+		Conditions:    conditions,
+		Actions:       actions,
+		PolicyErrors:  policyErrors,
+	}
+
+	log.Printf("Creating media retention policy %s", name)
+
+	policy, resp, err := pp.createPolicy(ctx, &reqBody)
+	log.Printf("Media retention policy creation status %#v", resp.Status)
+
+	if err != nil {
+		return diag.Errorf("Failed to create media retention policy %s: %s", name, err)
+	}
+
+	// Make sure form is properly created
+	policyId := policy.Id
+	d.SetId(*policyId)
+	log.Printf("Created media retention policy %s %s", name, *policy.Id)
+	return readMediaRetentionPolicy(ctx, d, meta)
+}
+
+// readMediaRetentionPolicy is used by the recording media retention policy resource to read a media retention policy from genesys cloud.
+func readMediaRetentionPolicy(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	pp := getPolicyProxy(sdkConfig)
+
+	log.Printf("Reading media retention policy %s", d.Id())
+
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		retentionPolicy, resp, err := pp.getPolicyById(ctx, d.Id())
+		if err != nil {
+			if gcloud.IsStatus404(resp) {
+				return retry.RetryableError(fmt.Errorf("failed to read media retention policy %s: %s", d.Id(), err))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read media retention policy %s: %s", d.Id(), err))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, gcloud.ResourceSurveyForm())
+		if retentionPolicy.Name != nil {
+			d.Set("name", *retentionPolicy.Name)
+		}
+		if retentionPolicy.Order != nil {
+			d.Set("order", *retentionPolicy.Order)
+		}
+		if retentionPolicy.Description != nil {
+			d.Set("description", *retentionPolicy.Description)
+		}
+		if retentionPolicy.Enabled != nil {
+			d.Set("enabled", *retentionPolicy.Enabled)
+		}
+		if retentionPolicy.MediaPolicies != nil {
+			d.Set("media_policies", flattenMediaPolicies(retentionPolicy.MediaPolicies, pp, ctx))
+		}
+		if retentionPolicy.Conditions != nil {
+			d.Set("conditions", flattenConditions(retentionPolicy.Conditions))
+		}
+		if retentionPolicy.Actions != nil {
+			d.Set("actions", flattenPolicyActions(retentionPolicy.Actions, pp, ctx))
+		}
+		if retentionPolicy.PolicyErrors != nil {
+			d.Set("policy_errors", flattenPolicyErrors(retentionPolicy.PolicyErrors))
+		}
```

Comment:
Same here with the resourcedata functions. To set a key with an interface array value returned from a function, you can use `SetNillableValueWithInterfaceArrayWithFunc`

Reply:
Used resourcedata util functions where applicable